block_writer/main.go

```diff
@@ -179,38 +188,60 @@ func main() {
 	writers := make([]blockWriter, *concurrency)
 
 	errCh := make(chan error)
+	var wg sync.WaitGroup
 	for i := range writers {
+		wg.Add(1)
 		writers[i] = newBlockWriter(db)
-		go writers[i].run(errCh)
+		go writers[i].run(errCh, &wg)
```

Comment:
can't adding to `wg` in `run` technically race since you also call `Wait` from a goroutine? Feels like you ought to be adding in the loop and only `Done`ing in the goroutine to be safe.